cpp/periodic_module/periodic_iterate.cpp

```diff
@@ -0,0 +1,241 @@
+  std::vector<std::string> relationship_names;
+  std::vector<std::string> primitive_names;
+};
+
+ParamNames ExtractParamNames(std::vector<std::string> columns, std::vector<mg::Value> &batch_row) {
+  std::vector<std::string> node_names;
+  std::vector<std::string> relationship_names;
+  std::vector<std::string> primitive_names;
+
+  for (size_t i = 0; i < columns.size(); i++) {
+    if (batch_row[i].type() == mg::Value::Type::Node) {
+      node_names.push_back(columns[i]);
+    } else if (batch_row[i].type() == mg::Value::Type::Relationship) {
+      relationship_names.push_back(columns[i]);
+    } else {
+      primitive_names.push_back(columns[i]);
+    }
+  }
+
+  return ParamNames{
+      .node_names = node_names, .relationship_names = relationship_names, .primitive_names = primitive_names};
+}
+
+std::vector<std::string> ConstructColumns(std::vector<std::string> columns, std::vector<mg::Value> &batch_row) {
+  std::vector<std::string> new_columns;
+
+  for (size_t i = 0; i < columns.size(); i++) {
+    if (batch_row[i].type() == mg::Value::Type::Node || batch_row[i].type() == mg::Value::Type::Relationship) {
+      new_columns.push_back(fmt::format("__{}_id", columns[i]));
+    } else {
+      new_columns.push_back(columns[i]);
+    }
+  }
+
+  return new_columns;
+}
+
+std::string ConstructQueryPreffix(ParamNames names) {
+  if (!names.node_names.size() && !names.relationship_names.size() && !names.primitive_names.size()) {
+    return "";
+  }
+
+  auto unwind_batch = fmt::format("UNWIND ${} AS {}", kBatchInternalName, kBatchRowInternalName);
+
+  std::vector<std::string> with_entity_vector;
+  for (const auto &node_name : names.node_names) {
+    with_entity_vector.push_back(fmt::format("{}.{} AS __{}_id", kBatchRowInternalName, node_name, node_name));
+  }
+  for (const auto &rel_name : names.relationship_names) {
+    with_entity_vector.push_back(fmt::format("{}.{} AS __{}_id", kBatchRowInternalName, rel_name, rel_name));
+  }
+  for (const auto &prim_name : names.primitive_names) {
+    with_entity_vector.push_back(fmt::format("{}.{} AS {}", kBatchRowInternalName, prim_name, prim_name));
+  }
+
+  auto with_variables = fmt::format("WITH {}", boost::algorithm::join(with_entity_vector, ", "));
+
+  std::string match_string = "";
+  std::vector<std::string> match_by_id_vector;
+  for (const auto &node_name : names.node_names) {
+    match_by_id_vector.push_back(fmt::format("MATCH ({}) WHERE ID({}) = __{}_id", node_name, node_name, node_name));
+  }
+  for (const auto &rel_name : names.relationship_names) {
+    match_by_id_vector.push_back(fmt::format("MATCH ()-[{}]->() WHERE ID({}) = __{}_id", rel_name, rel_name, rel_name));
+  }
+
+  if (match_by_id_vector.size()) {
+    match_string = boost::algorithm::join(match_by_id_vector, " ");
+  }
+
+  return fmt::format("{} {} {}", unwind_batch, with_variables, match_string);
+}
+
+mg::Map ConstructParams(std::vector<std::string> columns, std::vector<std::vector<mg::Value>> &batch) {
```

Comment:
Since this function is only used for constructing query params, I would say the better naming is ConstructQueryParams, what do you think?